include/rdm/param_data.h

```diff
@@ -0,0 +1,585 @@
+  rdmpd_data_type_t data_type;         /*!< The size of the data entries in the PD of the message for this PID. */
+  rdmpd_param_desc_cc_t command_class; /*!< Whether Get and or Set messages are implemented for the specified PID. */
+  rdmpd_sensor_unit_t unit;            /*!< The SI (International System of units) unit of the specified PID data. */
+  rdmpd_sensor_unit_prefix_t prefix;   /*!< The SI Prefix and multiplication factor of the units. */
+  uint32_t min_valid_value;            /*!< The lowest value the data can reach. */
+  uint32_t max_valid_value;            /*!< The highest value the data can reach. */
+  uint32_t default_value;              /*!< The default value of the data. */
+  RdmPdString description;             /*!< Describes the function of the specified PID. */
+} RdmPdParameterDescription;
+
+etcpal_error_t rdmpd_unpack_get_resp_parameter_description(const RdmParamData *pd,
+                                                           RdmPdParameterDescription *description_out);
+etcpal_error_t rdmpd_pack_get_resp_parameter_description(const RdmPdParameterDescription *description,
+                                                         RdmParamData *pd_out);
+
+// Get MANUFACTURER_LABEL See Section 10.5.4 of [RDM]
+etcpal_error_t rdmpd_unpack_get_resp_manufacturer_label(const RdmParamData *pd, RdmPdString *label_out);
+etcpal_error_t rdmpd_pack_get_resp_manufacturer_label(const RdmPdString *label, RdmParamData *pd_out);
+
+// Get DEVICE_MODEL_DESCRIPTION See Section 10.5.3 of [RDM]
+etcpal_error_t rdmpd_unpack_get_resp_device_model_description(const RdmParamData *pd, RdmPdString *description_out);
+etcpal_error_t rdmpd_pack_get_resp_device_model_description(const RdmPdString *description, RdmParamData *pd_out);
+
+// Get SOFTWARE_VERSION_LABEL See Section 10.5.9 of [RDM]
+etcpal_error_t rdmpd_unpack_get_resp_software_version_label(const RdmParamData *pd, RdmPdString *label_out);
+etcpal_error_t rdmpd_pack_get_resp_software_version_label(const RdmPdString *label, RdmParamData *pd_out);
+
+// Get/Set DEVICE_LABEL See Section 10.5.5 of [RDM]
+etcpal_error_t rdmpd_unpack_get_resp_device_label(const RdmParamData *pd, RdmPdString *label_out);
+etcpal_error_t rdmpd_pack_get_resp_device_label(const RdmPdString *label, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_device_label(const RdmParamData *pd, RdmPdString *label_out);
+etcpal_error_t rdmpd_pack_set_device_label(const RdmPdString *label, RdmParamData *pd_out);
+
+/* e1.33 Table A-15: RDM Parameter ID Defines */
+// Get/Set COMPONENT_SCOPE
+etcpal_error_t rdmpd_unpack_get_component_scope(const RdmParamData *pd, uint16_t *scope_slot_out);
+etcpal_error_t rdmpd_pack_get_component_scope(uint16_t scope_slot, RdmParamData *pd_out);
+
+/*! The maximum length of a scope string, including one extra byte for a null terminator. */
+#define RDMPD_MAX_SCOPE_STR_LEN 64
```

Comment:
Scope strings are actually a bit of a special case - per the standard, they are supposed to include a null terminator even on the wire. So this should actually be 63 since that's the field width in the message and it includes the null-term. This is not true of the rest of the string types.

Reply:
Adjusted to 63.